MekHQ/src/mekhq/MekHQ.java

```diff
@@ -654,6 +667,158 @@ public IconPackage getIconPackage() {
+     */
+    public void autoResolveConcluded(AutoResolveConcludedEvent autoResolveConcludedEvent) {
+        try {
+            String message = autoResolveConcludedEvent.controlledScenario() ?
+                "Your forces won the scenario. Did your side control the battlefield at the end of the scenario?" :
+                "Your forces lost the scenario. Do you want to declare your side as controlling the battlefield at the end of the scenario?";
+            String title = autoResolveConcludedEvent.controlledScenario() ? "Victory!" : "Defeat!";
+            boolean control = JOptionPane.showConfirmDialog(campaignGUI.getFrame(), message, title, JOptionPane.YES_NO_OPTION, JOptionPane.QUESTION_MESSAGE) == JOptionPane.YES_OPTION;
+            ResolveScenarioTracker tracker = new ResolveScenarioTracker(currentScenario, getCampaign(), control);
+            tracker.setClient(new SimulatedClient(autoResolveConcludedEvent.getGame(), getCampaign().getPlayer()));
+            tracker.setEvent(autoResolveConcludedEvent);
+            tracker.processGame();
+
+            ResolveScenarioWizardDialog resolveDialog =
+                new ResolveScenarioWizardDialog(campaignGUI.getFrame(),
+                    true, tracker);
+            resolveDialog.setVisible(true);
+            if (resolveDialog.wasAborted()) {
+                // I think I can get rid of this, but I'm not sure atm, testing
+//                for (UUID personId : tracker.getPeopleStatus().keySet()) {
+//                    Person person = getCampaign().getPerson(personId);
+//                    person.setHits(person.getHitsPrior());
+//                }
+                return;
+            }
+            PostScenarioDialogHandler.handle(
+                campaignGUI, getCampaign(), (AtBScenario) currentScenario, tracker, autoResolveConcludedEvent.controlledScenario());
+        } catch (Exception ex) {
+            logger.error("Error during auto resolve concluded", ex);
+        }
+    }
+
+
+    private static class SimulationScore {
+        private final AtomicInteger victories;
+        private final AtomicInteger losses;
+        private final AtomicInteger draws;
+        private final AtomicInteger gamesRun;
+
+        public SimulationScore() {
+            this.victories = new AtomicInteger(0);
+            this.losses = new AtomicInteger(0);
+            this.draws = new AtomicInteger(0);
+            this.gamesRun = new AtomicInteger(0);
+        }
+
+        public void addResult(AutoResolveConcludedEvent event) {
+            this.addResult(event.getGame());
+        }
+
+        public void addResult(IGame game) {
+            if (game.getVictoryTeam() == 1) {
+                victories.incrementAndGet();
+            } else if (game.getVictoryTeam() == 2) {
+                losses.incrementAndGet();
+            } else {
+                draws.incrementAndGet();
+            }
+            gamesRun.incrementAndGet();
+        }
+
+        public int getVictories() {
```

Comment:
Not in this case, since it is counting losses, victories, draws, this is very short and simple, and the object itself is also very direct on its responsibiliites `SimulationScore`.